### PR TITLE
bug 1727318: pkg/clusterconfig: use infra.Status.Platform if PlatformStatus is nil

### PR DESCRIPTION
**Pull request by runcom**
As witnessed in https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade-rollback-4.1-to-4.2/16 - when moving from 4.1 to 4.2 we need to take care of the missing `infra.Status.PlatformStatus` (introduced only in 4.2) or we get a panic (https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade-rollback-4.1-to-4.2/16/artifacts/e2e-aws-upgrade/pods/openshift-image-registry_cluster-image-registry-operator-7dfbcc46df-f4z68_cluster-image-registry-operator.log)

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by bparees** [comment]
/lgtm

@coreydaley @dmage fyi this was causing a failure in the 4.1->4.2 upgrade jobs.

File: pkg/clusterconfig/clusterconfig.go [hunk 1]
```diff
@@ -91,7 +91,11 @@ func GetAWSConfig(kubeconfig *rest.Config, listers *regopclient.Listers) (*Confi
 		return nil, err
 	}
 
-	if infra.Status.PlatformStatus.Type == configapiv1.AWSPlatformType {
+	platformType := infra.Status.Platform
+	if infra.Status.PlatformStatus != nil {
+		platformType = infra.Status.PlatformStatus.Type
+	}
+	if platformType == configapiv1.AWSPlatformType {
```
Comment (coreydaley) [inline pkg/clusterconfig/clusterconfig.go]: If infra.Status.PlatformStatus is nil, won't line 99 fail with a nil pointer?
`cfg.Storage.S3.Region = infra.Status.PlatformStatus.AWS.Region`

Comment (runcom) [reply pkg/clusterconfig/clusterconfig.go]: ouch

Comment (runcom) [reply pkg/clusterconfig/clusterconfig.go]: fixing

Comment (bparees) [reply pkg/clusterconfig/clusterconfig.go]: i think you're correct, plus the other platformstatus usages you idenfied in slack will also be a problem

Comment (runcom) [reply pkg/clusterconfig/clusterconfig.go]: fixed by reintroducing the install config inspection

**Comment by runcom** [comment]
I'm actually going to close this one cause of https://github.com/openshift/cluster-image-registry-operator/pull/312

**Pull request closed without merging**